sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -292,10 +292,18 @@ case class TruncateTableCommand(
     } else if (catalog.isTemporaryTable(tableName)) {
       logError(s"table '$tableName' in TRUNCATE TABLE is a temporary table.")
     } else {
+      val table = catalog.getTableMetadata(tableName)
+      if (table.tableType == VIEW ) {
+        throw new AnalysisException(
+          s"TRUNCATE TABLE is not allowed on a view: ${table.qualifiedName}")
```

Comment:
with that format you can just use `assertUnsupported` in `DDLSuite` when you add a test for each of these cases